lib/util/batch.js

```diff
@@ -121,7 +121,25 @@ function supportsCql31(version) {
   return version.major > 3 || (version.major === 3 && version.minor >= 1);
 }
 
-var endsWithSemicolonRegex = new RegExp(/\s*;\s*$/);
+function containsDeleteOrUpdateStatement(query) {
+  var match = false;
+  if (query instanceof Batch) {
+    _.each(query.context.queries, function (subQuery) {
+      if (match) { return; }
+      if (containsDeleteOrUpdateStatement(subQuery)) {
+        match = true;
+      }
+    });
+  } else {
+    if (containsDeleteOrUpdateRegex.test(query.context.cql)) {
+      match = true;
+    }
+  }
+  return match;
+}
+
+var containsDeleteOrUpdateRegex = /(?:update|delete)\s+[^\s,\(\)]/i;
```

Comment:
I think this could give you a false positive for certain strings a user might input. It should probably be 
/^\s*(?:update|delete)\s+[^\s,\(\)]/i